airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py

```diff
@@ -664,6 +666,57 @@ def _flat_associations(self, records: Iterable[MutableMapping]) -> Iterable[Muta
             yield record
 
 
+class SemiIncrementalStream(Stream, IncrementalMixin):
```

Comment:
nit: a better name for this pattern is `ClientSideIncrementalStream`

Reply:
renamed